### PR TITLE
Always upgrade pip on creating virtualenv

### PR DESCRIPTION
**Pull request by kosmos342**
Some packages which doesn't have `name` in `setup.py` can't be installed by old versions of pip (I didn't check from what version). Pip names them as `unknown`

for example:
[mock==3.0.5](https://github.com/testing-cabal/mock/blob/e0180b98d0e07e895a3f699b7e9afcac4716fc03/setup.py#L6-L10)
[pytest==4.6.3](https://github.com/pytest-dev/pytest/blob/b8e65d03bf563d5de3a285af187fd00a2f202e5e/setup.py#L24-L40)

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by kosmos342** [comment]
looks like tests fails for master too https://jenkins.yola.net/job/Test-yodeploy/284/JENKINS_SLAVE=trusty/consoleFull

**Comment by RayeN** [comment]
@stefanor, we need your opinion regarding this.

File: yodeploy/virtualenv.py [hunk 1]
```diff
@@ -264,4 +261,4 @@ def remove_fragile_symlinks(ve_dir):
             continue
 
         log.debug('Removing fragile symlink %s -> %s', path, dest)
-        os.unlink(path)
+        os.unlink(path)
```
Comment (snitch) [inline yodeploy/virtualenv.py]: W292 no newline at end of file

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: yodeploy/virtualenv.py [hunk 2]
```diff
@@ -72,18 +72,13 @@ def create_ve(
...
     # The venv module makes a lot of our reclocateability problems go away, so
     # we only use the virtualenv library on Python 2.
     if python_version.startswith('3.'):
         subprocess.check_call((
             'python%s' % python_version, '-m', 'venv', ve_dir))
-        pip_version = subprocess.check_output((
-            ve_python, '-c', 'import ensurepip; print(ensurepip.version())'))
-        if parse_version(pip_version) < parse_version('9'):
-            pip_install(ve_dir, pypi, '-U', 'pip')
         pip_install(ve_dir, pypi, 'wheel')
```
Comment (stefanor) [inline yodeploy/virtualenv.py]: Move this after the pip upgrade, so that it is cached?

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,12 @@
 Change Log
 ==========
 
+0.10.2
+-----
+
+* Always upgrade `pip` for Python virtualenvs.
+* Install `wheel` in all Python virtualenvs.
```
Comment (stefanor) [inline CHANGELOG.md]: We were expecting it to be installed under Python 2, before. So I suspect this change is essentially a no-op for Python 2 & 3.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: CHANGELOG.md (hunk 3 above)
Comment (kosmos342) [reply CHANGELOG.md]: Is it ok now?

Comment (stefanor) [reply CHANGELOG.md]: I don't think that's very clear: How about: Explicitly install `wheel` after upgrading pip.

Or just have no changelog entry for this.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Pull request merged**